lib/TPP/TensorInit.cpp

```diff
@@ -70,29 +70,35 @@ DenseElementsAttr TensorInit::get(ShapedType shape) {
 
 void TensorInit::insert(size_t index, float value) {
   buffer[index] = llvm::APFloat(value);
-  if (type == DataType::BF16)
-    toBF16(buffer[index]);
+  convertType(buffer[index]);
 }
 
 void TensorInit::push(float value) {
   buffer.push_back(llvm::APFloat(value));
-  if (type == DataType::BF16)
-    toBF16(buffer.back());
+  convertType(buffer.back());
 }
 
-float TensorInit::at(size_t index) {
-  return buffer[index].convertToFloat();
+void TensorInit::convertType(llvm::APFloat &value) {
+  switch (type) {
```

Comment:
Perhaps a check if the type is already correct? I'm guessing the cast is more expensive than a check?